docs: where to start (#5941)
vuestorefront/vue-storefront: packages/core/docs/general/where-to-start.md

# Where to start?

This document will show where you should start if you are:

[[toc]]

## Considering using Vue Storefront

Vue Storefront is a **platform-agnostic e-commerce PWA frontend framework** that can work with any eCommerce backend API. Sounds great, right? But that is not all.

With our product, you can easily solve the most common problems in the eCommerce e.g.
- [Long time to market](/#long-time-to-market)
- [Slow, unresponsive online shop](/#slow-unresponsive-online-shop)
- [Unwieldy architectural decisions](/#unwieldy-architectural-decisions)
- [Painful or impossible migrations](/#painful-or-impossible-migrations)
- [Lack of platform-specific competencies](/#lack-of-platform-specific-competencies)
- [Lack of flexibility](/#lack-of-flexibility)

Also, we are proud of:
- 300+ [Live Implementations](https://www.vuestorefront.io/live-projects)
- 100+ [Agency Partners](https://www.vuestorefront.io/partner-agencies)

> - 4500+ [Community members](http://discord.vuestorefront.io/)

Reviewer: ```suggestion
- active [Discord community](http://discord.vuestorefront.io/)
```

Since our move to Discord, this number is much lower (getting close to 1000)

You should definitely visit [Introduction](/) or [Key conepts](/general/key-concepts.html) section for more details about Vue Storefront. 

**Introduction** section will guide you through:
- [What is Vue Storefront?](/#what-is-vue-storefront)
- [Problems Vue Storefront solves](/#problems-vue-storefront-solves)
- [eCommerce integrations](/#ecommerce-integrations)
- [Tech stack](/#tech-stack)
- [What's next](/#what-s-next)

**Key concepts** section will guide you through:
- [Progressive technology](/general/key-concepts.html#progressive-technology)
- [Reusability](/general/key-concepts.html#reusability)
- [Extendability and custom integrations](/general/key-concepts.html#extendability-and-custom-integrations)
- [Easy configuration](/general/key-concepts.html#easy-configuration)
- [Backend-agnostic](/general/key-concepts.html#backend-agnostic)

## Developer creating a shop

> As a developer, you will be creating a shop from scratch to ready for production. Vue Storefront aims to meet the expectations of developers and guide you helping smoothly through all these steps.

Reviewer: > you will be creating a shop from scratch

This isn't true. We created VSF so that people don't have to create shops from scratch :)

> At the start, you should familiarize yourself with our **Guides** section. Includes all that you need to know for successful shop implementation. Also, we have a section with the most advanced topics.

Reviewer: ```suggestion
You should start by familiarizing yourself with the **Guides** category. It includes all you need to know to create a shop successfully. If you want to dive deeper, check out the **Advanced** category.
```

**Guides** section will guide you through:
- [Theme](../guide/theme.html)
- [Configuration](../guide/configuration.html)
- [Composables](../guide/composables.html)
- [Getters](../guide/getters.html)
- [Product catalog](../guide/product-catalog.html)
- [Authentication](../guide/authentication.html)
- [User profile](../guide/user-profile.html)
- [Cart and Wishlist](../guide/cart-and-wishlist.html)
- [Checkout](../guide/checkout.html)

**Advanced** section will guide you through:
- [Architecture](../advanced/architecture.html)
- [Application context](../advanced/context.html)
- [Calling platform API](../advanced/calling-platform-api.html)
- [Extending GraphQL Queries](../advanced/extending-graphql-queries.html)
- [Server Middleware](../advanced/server-middleware.html)
- [Internationalization](../advanced/internationalization.html)
- [Performance](../advanced/performance.html)
- [SSR Cache](../advanced/ssr-cache.html)
- [Logging](../advanced/logging.html)
- [API Reference](../core/api-reference/)

> To be up to date, we recommend to following [Migration guide](../migrate) and Vue Storefront 2 [Roadmap](https://www.notion.so/vuestorefront/Vue-Storefront-2-Next-High-level-Roadmap-201cf06abb314b84ad01b7b8463c0437) .

Reviewer: ```suggestion
To be up to date, we recommend following our [Roadmap](https://www.notion.so/vuestorefront/Vue-Storefront-2-Next-High-level-Roadmap-201cf06abb314b84ad01b7b8463c0437) and joining our [Discord community](http://discord.vuestorefront.io/).
```

Vue Storefront integrates with a wide range of technologies and each has its own needs, so there is no generic tutorial for all of them.

However, depending on the kind of integrations you want to build, you should familiarize yourself with **Building Integration** section. Especially integration guide for [eCommerce](../integrate/integration-guide.html) and [CMS](../integrate/cms.html) platforms.

**eCommerce** section will guide you through: 
- [Introduction](../integrate/integration-guide.html#introduction)
- [Scope](../integrate/integration-guide.html#scope)
- [Getting started](../integrate/integration-guide.html#getting-started)
- [Creating an api-client](../integrate/integration-guide.html#creating-an-api-client)
- [Creating composables](../integrate/integration-guide.html#creating-composables)
- [Creating getters](../integrate/integration-guide.html#creating-getters)
- [Creating a theme](../integrate/integration-guide.html#creating-a-theme)

**CMS** section will guide you through:
- [configuration / content fetching / content rendering](../integrate/cms.html#what-is-needed)

> - [usage example in the real application](../integrate/cms.html#usage-example-in-the-real-application)

Reviewer: Remove these lists. IMHO the links above are enough. We don't need to list each section.

## Designer creating a theme

> Vue Storefront default theme is mostly based on Single UI components turned into a powerful design system for e-commerce we called [Storefront UI](https://www.storefrontui.io/)

Reviewer: ```suggestion
Our default theme is mostly based on a powerful design system for e-commerce called [Storefront UI](https://www.storefrontui.io/). If you don't want to use it, you can replace it with any other UI library or custom components, by changing the templates in Vue files.
```

> If you don't want to use [Storefront UI](https://www.storefrontui.io/) (it's just a UI layer), the project can work with any other UI library or a custom code. So it totally depends on you - create your own theme, edit the existing one or use our default theme.

Reviewer: ```suggestion
```

At the start you should familiarize yourself with the specially prepared [theme](../guide/theme.html) guide.

**Theme** section will guide you through:
- [Directory structure](../guide/theme.html#directory-structure)
- [Storefront UI](../guide/theme.html#directory-structure)

> - [How to customizing the theme](../guide/theme.html#customizing-the-theme)

Reviewer: ```suggestion
- [How to customize the theme](../guide/theme.html#customizing-the-theme)
```

- [Routing](../guide/theme.html#routing)

> - [How to updating styles](../guide/theme.html#updating-styles)

Reviewer: ```suggestion
- [How to update styles](../guide/theme.html#updating-styles)
```

# Where to start?

This document will show where you should start if you are:

[[toc]]

[comment]: <> (## Developer creating a shop)

## Integrator

All integrations created and maintained by Core Team and our partners you can find on the special created [integrations](../integrations/) page.

> Didn't find the integration you were looking for? You can create your own eCommerce, CMS, or other integrations with Vue Storefront but remember about:
> 
> ::: warning Want to build an integration?
> If you want to integrate with Vue Storefront, please **contact the core team on our [Discord](https://discord.vuestorefront.io) server** first. We are eager to help you with building it and ensuring its high quality! Building the integration with the help of the core team is the best way to keep its quality high and making it officially recommended once completed.
> :::

Reviewer: This warning is already on the pages for creating integrations, so there is no need to have it here as well.

```suggestion
If you didn't find the integration you were looking for, you can create your own.
```

> Depending on what kind of integrations you want to build you should familiarize yourself with **Building Integration** section. Especially integration guide for [eCommerce](../integrate/integration-guide.html) and [CMS](../integrate/cms.html).

Reviewer: ```suggestion
Depending on the kind of integrations you want to build, you should familiarize yourself with **Building Integration** section. Especially integration guide for [eCommerce](../integrate/integration-guide.html) and [CMS](../integrate/cms.html) platforms.
```

**eCommerce** section will guide you through: 
- [introduction](../integrate/integration-guide.html#introduction)
- [scope](../integrate/integration-guide.html#scope)
- [getting started](../integrate/integration-guide.html#getting-started)

> - [how to creating an api-client](../integrate/integration-guide.html#creating-an-api-client)
> - [how to creating composables](../integrate/integration-guide.html#creating-composables)
> - [how to creating getters](../integrate/integration-guide.html#creating-getters)
> - [how to creating a theme](../integrate/integration-guide.html#creating-a-theme)

Reviewer: ```suggestion
- [creating an api-client](../integrate/integration-guide.html#creating-an-api-client)
- [creating composables](../integrate/integration-guide.html#creating-composables)
- [creating getters](../integrate/integration-guide.html#creating-getters)
- [creating a theme](../integrate/integration-guide.html#creating-a-theme)
```
